oc-chef-pedant/lib/pedant/rspec/cookbook_util.rb

```diff
@@ -330,44 +373,75 @@ def new_cookbook(name, version, opts = {})
             "recipes" => opts[:meta_recipes] || {}
           }
         }
+        if opts.key?(:payload)
+          opts[:payload].each do |part, files|
+            cb[part.to_s] ||= []
+            cb[part.to_s] += files
+          end
+        end
+        cb
       end
 
-      shared(:default_description) { "A fabulous new cookbook" }
-      shared(:default_long_description) { "" }
-      shared(:default_maintainer) { "Your Name" }
-      shared(:default_maintainer_email) { "youremail@example.com" }
-      shared(:default_license) { "Apache v2.0" }
+      def new_cookbook_v2(name, version, opts = {})
+        all_files = []
+        if opts.key?(:recipes)
+          all_files = opts[:recipes].each_with_object([]) do |r, acc|
+            af = r.dup
+            af["name"] = "recipes/#{r['name']}"
+            acc << af
+          end
+        end
+
+        if opts.key?(:payload)
+          opts[:payload].each do |part, files|
+            all_files += files.each_with_object([]) do |f, acc|
+              file = f.dup
+              file["name"] = "#{part}/#{file["name"]}"
```

Comment:
Looking at the usage: https://github.com/chef/chef-server/pull/1034/files#diff-d5911d79b4d2796916287911d6a428fcR69 it seems like we'd need to do something different for root_files?